hkdf.go

```diff
@@ -119,6 +122,18 @@ func ExtractHKDF(h func() hash.Hash, secret, salt []byte) ([]byte, error) {
 		return nil, err
 	}
 
+	// If calling code specifies nil salt, replace it with a buffer of hashLen
+	// zeros, as specified in RFC 5896 and as OpenSSL EVP_KDF-HKDF documentation
+	// instructs. Take a slice of a preallocated buffer to avoid allocating new
+	// buffer per call.
+	if len(salt) == 0 {
+		hlen := h().Size()
+		if hlen > len(hkdfAllZerosSalt) {
+			return nil, errors.New("nil salt is not supported for the specified hash algorithm")
```

Comment:
This seems a little vague to me--the message seems to imply something is fundamentally incompatible, but "our internal buffer isn't long enough" doesn't seem fundamental. E.g. if someone wants to pass nil and use a hash function that requires a salt longer than 64 bytes, it would be ok for them to send a PR to golang-fips/openssl to increase the size of this buffer, right?

Also, "nil salt" could use an update. Here's an attempt:

```suggestion
			return nil, errors.New("caller requested a zero-filled buffer as salt by passing zero-length salt input, but the internal buffer is not large enough for the specified hash algorithm")
```